stdlib/src/collections/inline_list.mojo

```diff
@@ -245,5 +246,5 @@ struct InlineList[ElementType: CollectionElementNew, capacity: Int = 16](Sized):
             value: The value to append.
         """
         debug_assert(self._size < capacity, "List is full.")
-        self._array[self._size] = value^
+        self._array[self._size] = UnsafeMaybeUninitialized[ElementType](value^)
```

Comment:
**Question** Does this trigger the `abort()` in `UnsafeMaybeUninitialized.__moveinit__()`?

I wonder if this would be better:

```suggestion
        self._array[self._size].write(value^)
```

Reply:
The `InlineList` is very well tested. The `append` function too. So if this were to trigger an abort(), we would have seen it in the CI. For example there: https://github.com/modularml/mojo/blob/nightly/stdlib/test/collections/test_inline_list.mojo#L49

I believe there is no `__moveinit__` called on the `UnsafeMaybeUninitialized` because in Mojo, the `__init__` method always build the result "in-place", it's similar to return value optimization, thus there is no move.

The suggestion you made is shorter, so sure, I don't mind using it.